partner_preview_scripts/ValidationLab.py

```diff
@@ -0,0 +1,384 @@
+    projectDescription = parameters['projectDescription']
+    diseaseFromSource = parameters['diseaseFromSource']
+    diseaseFromSourceMapId = parameters['diseaseFromSourceMappedId']
+    confidenceCutoff = parameters['confidenceCutoff']
+    cellLineFile = parameters['cellLineFile']
+    hypothesisFile = parameters['hypothesisFile']
+
+    # Cell line data:
+    # Reading cell metadata from validation lab:
+    validation_lab_cell_lines = (
+        spark.read.csv(cellLineFile, sep='\t', header=True)
+
+        # Renaming columns:
+        .withColumnRenamed('cellLines', 'name')
+
+        # Joining dataset with cell model data read downloaded from Sanger website:
+        .join(cellPassportDf, on='name', how='left')
+
+        # Adding UBERON code to tissues (it's constant colon)
+        .withColumn('tissueID', lit('UBERON_0000059'))
+
+        # generating disease cell lines object:
+        .withColumn(
+            'diseaseCellLines',
+            array(struct(col('name'), col('id'), col('tissue'), col('tissueId')))
+        )
+        .drop(*['id', 'tissue', 'tissueId'])
+        .persist()
+    )
+
+    logging.info(f'Validation lab cell lines has {validation_lab_cell_lines.count()} cell types.')
+
+    # Defining how to process biomarkers:
+    # 1. Looping through all possible biomarker - from biomarkerMaps.keys()
+    # 2. The biomakers are then looked up in the map and process based on how the map defines.
+    # 3. Description is also added read from the map.
+    expressions = map(
+        # Function to process biomarker:
+        lambda biomarker: (biomarker, get_biomarker(
+            lit(biomarker), col(biomarker))),
+
+        # Iterator to apply the function over:
+        BIOMARKERMAPS.keys()
+    )
+
+    # Applying the full map on the dataframe one-by-one:
+    biomarkers = reduce(lambda DF, value: DF.withColumn(*value), expressions, validation_lab_cell_lines)
+
+    # The biomarker columns are unstacked into one single 'biomarkers' column:
+    biomarker_unstack = f'''stack({len(BIOMARKERMAPS.keys())}, {", ".join([f"'{x}', {x}" for x in BIOMARKERMAPS.keys()])}) as (biomarker_name, biomarkers)'''
+
+    validation_lab_cell_lines = (
+        biomarkers
+
+        # Selecting cell line name, cell line annotation and applyting the stacking expression:
+        .select('name', 'diseaseCellLines', expr(biomarker_unstack))
+
+        # Filter out all null biomarkers:
+        .filter(col('biomarkers').isNotNull())
+
+        # Grouping data by cell lines again:
+        .groupBy('name')
+        .agg(
+            collect_list('biomarkers').alias('biomarkers'),
+            first(col('diseaseCellLines')).alias('diseaseCellLines')
+        )
+    )
+
+    # Reading and processing hypothesis data:
+    hypothesis = (
+        spark.read.csv(hypothesisFile, sep='\t', header=True)
+        .withColumn('hypothesis', parse_hypothesis(col('biomarker'), col('biomarkerStatus'), col('description')))
```

Comment:
Perhaps the data is expected to change and become more complex, but the current implementation can be achieved directly with Spark:

```
.withColumn(
    'hypothesis', 
    struct(
        concat(col('biomarker'), lit('-'), col('biomarker_status')).alias('hypothesis'),
        col('hypothesis').alias('description')
    )
)
```